diff --git a/node_modules/expo-document-picker/build/ExpoDocumentPicker.js b/node_modules/expo-document-picker/build/ExpoDocumentPicker.js
index d887611..1e9676b 100644
--- a/node_modules/expo-document-picker/build/ExpoDocumentPicker.js
+++ b/node_modules/expo-document-picker/build/ExpoDocumentPicker.js
@@ -1,3 +1,40 @@
 import { requireNativeModule } from 'expo-modules-core';
-export default requireNativeModule('ExpoDocumentPicker');
+import { Platform } from 'react-native';
+let ExpoDocumentPicker;
+if (Platform.OS === 'harmony') {
+    const DocumentPicker = require('react-native-document-picker').default;
+    ExpoDocumentPicker = {
+        getDocumentAsync: async ({ type = ['*/*'], copyToCacheDirectory = true, multiple = false, }) => {
+            try {
+                const results = await DocumentPicker.pick({
+                    type,
+                    allowMultiSelection: multiple,
+                    ...(copyToCacheDirectory ? { copyTo: 'cachesDirectory' } : {}),
+                });
+                return {
+                    canceled: false,
+                    assets: results.map((item) => ({
+                        name: item.name || '',
+                        size: item.size ?? undefined,
+                        uri: item.fileCopyUri || item.uri,
+                        mimeType: item.type || undefined,
+                    })),
+                };
+            }
+            catch (error) {
+                if (DocumentPicker.isCancel(error)) {
+                    return {
+                        canceled: true,
+                        assets: null,
+                    };
+                }
+                throw error;
+            }
+        },
+    };
+}
+else {
+    ExpoDocumentPicker = requireNativeModule('ExpoDocumentPicker');
+}
+export default ExpoDocumentPicker;
 //# sourceMappingURL=ExpoDocumentPicker.js.map
